### PR TITLE
Added typescript typings in index.d.ts.

### PR DESCRIPTION
**Pull request by DJWassink**
I am using property-validator in a typescript codebase so I had to make some typings. Guess it would be usefull for some other people hence this pull request.

Let me know what you think and if we want this 😄

### CODE REVIEW

**Comment by nettofarah** [comment]
This is just beautiful, @DJWassink! 😍
I'll take a more in-depth look sometime tonight, but I'm incredibly excited about this.

@bjrmatos Can I get your 👀 on this PR?

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    //result of a validation function
+    export type ValidatorResult = {
+        field: string;
+        message: string;
+        result: boolean;
+    }
+
+    //request validators
+    export function validate(params: string[], validations: Validator[]): ValidateResult;
```
Comment (nettofarah) [inline index.d.ts]: Shouldn't params here be an object?

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    export function validateParams(request: { params: {} }, validations: Validator[]): ValidateResult;
+    export function validateQuery(request: { query: {} }, validations: Validator[]): ValidateResult;
+    export function validateBody(request: { body: {} }, validations: Validator[]): ValidateResult;
+    export function validateHeaders(request: { headers: {} }, validations: Validator[]): ValidateResult;
+    export function validateAll(request: { query?: {}, params?: {}, body?: {} }, validations: Validator[]): ValidateResult;
+
+    //request assertions
+    export function ensure(properties: {}, validations: Validator[], validationStrategy: () => ValidateResult): ValidateResult;
```
Comment (DJWassink) [inline index.d.ts]: Good catch!

File: index.d.ts (hunk 1 above)
Comment (DJWassink) [inline index.d.ts]: Wowha don't know how I got that one messed up :laughing:

**Comment by nettofarah** [comment]
@DJWassink You might wanna start another PR and push something to the examples folder :)

**Comment by DJWassink** [comment]
Well that's the funny thing. For users it looks exactly the same. Only thing is, if they put for example a `string array` instead of a `object` (hehe) in for example the `validate` function a developer its IDE or their build tool would tell them the TS code can't be compile since your are not implementing the correct interface.

**Comment by nettofarah** [comment]
That's what I figured.
I was mostly curious to see some autocompleting in action 😅

**Comment by nettofarah** [comment]
@DJWassink I gave you push access to the repo btw. Feel free to come back with more PRs anytime :)

**Comment by DJWassink** [comment]
Aah right ofcourse :tada: 

I use WebStorm which in my opinion isn't all that strong with autocompletion. Maybe VS Code is better in this or that I need to tweak my settings a bit. Anyhow this is what for example shows up while typing:

screenshot_2017-04-11_09-48-35

**Comment by DJWassink** [comment]
Ooh and something I completely forgot, which is my workflow. If I wan't to know how a method should be called I usually ctrl+enter while my cursor is over a function and my editor opens the defenition files which show all the parameter types etc.

**Comment by DJWassink** [comment]
@nettofarah I am going to open a new PR pretty soon with some fixes for some typings so if you could hold off a new release for a while that would be great :+1:

**Comment by DJWassink** [comment]
@nettofarah Take a look at #17 ;)

**Pull request merged**